### PR TITLE
Add Telemetry module to ADS

### PR DESCRIPTION
**Pull request by liudmylaru**
### Description

First step to include telemetry to aqua module is to move telemetry module into ADS.
https://jira.oci.oraclecorp.com/browse/ODSC-53773

### What was done

- move telemetry module from odsc_cli to ADS codebase

### Validation

- builded whl
- installed it in nb-session
- run code to record event in the conda bucket (I am not sure if aqua logs created in dev)
```
from ads import set_auth
from ads.telemetry.client import TelemetryClient

set_auth("resource_principal")
AQUA_BUCKET = "service-managed-models"
AQUA_BUCKET_NS = "ociodscdev"
telemetry = TelemetryClient(bucket=AQUA_BUCKET, namespace=AQUA_BUCKET_NS)
category = "ads/telemetry/module"
action = "test"
telemetry.record_event_async(category=f"{category}", action=action)
```
- checked logs records. Saw an expected entry with. check the ticket (https://jira.oci.oraclecorp.com/browse/ODSC-53773) comment section fro a screenshot.

### CODE REVIEW

File: ads/telemetry/base.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        Parameters
+        ----------
+        bucket : str
+            OCI object storage bucket name storing the telemetry objects.
+        namespace : str, optional
+            Namespace of the OCI object storage bucket, by default None.
+        """
+        if OCI_RESOURCE_PRINCIPAL_VERSION:
+            set_auth("resource_principal")
+        self._auth = default_signer()
```
Comment (VipulMascarenhas) [inline ads/telemetry/base.py:30]: I think we need to pass service endpoint here. Before, when trying to hit int endpoints, auth was point to prod when this wasn't set.

`self._auth = default_signer({"service_endpoint": OCI_ODSC_SERVICE_ENDPOINT})`

**Review by VipulMascarenhas — APPROVED** [review]
changes look good to me 👍

File: ads/telemetry/base.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        Parameters
+        ----------
+        bucket : str
+            OCI object storage bucket name storing the telemetry objects.
+        namespace : str, optional
+            Namespace of the OCI object storage bucket, by default None.
+        """
+        if OCI_RESOURCE_PRINCIPAL_VERSION:
+            set_auth("resource_principal")
+        self._auth = default_signer({"service_endpoint": OCI_ODSC_SERVICE_ENDPOINT})
```
Comment (qiuosier) [inline ads/telemetry/base.py:30]: We should not use ODSC endpoint here as this is for object storage.

File: ads/telemetry/base.py (hunk 1 above)
Comment (VipulMascarenhas) [reply ads/telemetry/base.py:30]: just saw QQ's comment, I didn't know this only works for DS client and not other clients. :)

**Review by VipulMascarenhas — APPROVED** [review]

**Pull request merged**